src/transport/MessageHeader.h

```diff
@@ -185,14 +185,14 @@ class PacketHeader
         }
         else
         {
-            mFlags.value &= ~Header::Flags::kDestinationNodeIdPresent;
+            mFlags.value = static_cast<uint16_t>(mFlags.value & ~Header::Flags::kDestinationNodeIdPresent);
         }
         return *this;
     }
 
     PacketHeader & ClearDestinationNodeId()
     {
-        mFlags.value &= ~Header::Flags::kDestinationNodeIdPresent;
+        mFlags.value = static_cast<uint16_t>(mFlags.value & ~Header::Flags::kDestinationNodeIdPresent);
```

Comment:
this seems really unfortunate, is this an artifact of the size of the constants?

Reply:
I'm running this locally and the static cast (and reorg) doesn't seem to be necessary

Reply:
It's necessary afaik, because `~foo` is of type `int` if `foo` is a type smaller than `int`....  Certainly I had build bustage both in CI and locally without that change.

Reply:
my test was `./gn_build.sh`, perhaps not exhaustive

Reply:
That's my test too... if I apply this patch, then specifically back out this part:
```
--- a/src/transport/MessageHeader.h
+++ b/src/transport/MessageHeader.h
@@ -193,5 +193,5 @@ public:
     PacketHeader & ClearDestinationNodeId()
     {
-        mFlags.value = static_cast<uint16_t>(mFlags.value & ~Header::Flags::kDestinationNodeIdPresent);
+        mFlags.value &= ~Header::Flags::kDestinationNodeIdPresent;
         mDestinationNodeId.ClearValue();
         return *this;
```
I get this when running `./gn_build.sh`:
```
In file included from ../../src/transport/Base.h:32,
                 from ../../src/transport/RendezvousParameters.h:21,
                 from ../../src/transport/RendezvousSession.h:28,
                 from ../../src/controller/CHIPDeviceController.h:35,
                 from ../../src/controller/CHIPDeviceController.cpp:32:
../../src/transport/MessageHeader.h: In member function 'chip::PacketHeader& chip::PacketHeader::ClearDestinationNodeId()':
../../src/transport/MessageHeader.h:195:22: error: conversion from 'int' to 'uint16_t' {aka 'short unsigned int'} may change value [-Werror=conversion]
  195 |         mFlags.value &= ~Header::Flags::kDestinationNodeIdPresent;
      |         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Oh, I get that when it's compiling the nrf5 lock app.  I do have `NRF5_SDK_ROOT` set, so `./gn_build.sh` is compiling the nrf5 bits.  It's possible that's the difference.

Reply:
The whole point of a struct and constexpr was to avoid this :( sigh.

Reply:
Maybe add an issue for me to add a helper to the Flags struct. This should be more readable:

```
mFlags.ClearFlag(Header::Flags::kDestinationNodeIdPresent);
mFlags.SetFlag(Header::Flags::kSourceNodeIdPresent)
```

and we will have the uglyness only in one place

Reply:
Good idea.  Filed https://github.com/project-chip/connectedhomeip/issues/2823